### PR TITLE
feat: align Codex classification and add regression tests after #609

### PR DESCRIPTION
**Pull request by hatayama**
## Summary

PR #609 introduced project-level Codex configuration support — a great improvement that aligns Codex with other editors. This follow-up PR completes the transition by updating the remaining classification and documentation to match, and adds regression tests for long-term maintainability.

### 1. Align Codex Classification (chore)
- **McpServerConfigFactory.cs**: Moved Codex from desktop (absolute path) to CLI/project-level (relative path) category
- **UnityMcpPathResolver.cs**: Updated doc comments to list Codex under relative-path editors
- **CodexTomlConfigService.cs**: Updated class comment from ~/.codex to {projectRoot}/.codex
- **CodexTomlConfigService.cs**: Renamed BuildBlock parameter serverAbsolutePath → serverPath (reflects actual usage)

### 2. Add Regression Tests (test)
- **CodexConfigTests.cs**: 12 test cases covering:
  - BuildBlock/ReadCurrentValues round-trip with forward/backslashes
  - NormalizeForCompare edge cases
  - MakeRelativeToConfigurationRoot path conversion
  - GetCodexConfigPath resolves under project root (not home dir)
  - McpServerConfigFactory correctly classifies Codex as relative-path editor

## Test Results

All 12 tests pass:
- Round-trip: relative paths with forward/backward slashes
- Normalization: backslash → forward slash conversion
- Path resolution: project-relative vs absolute fallback
- Config path: project root vs home directory
- Factory classification: Codex uses relative paths

## Migration Note for Existing Users

Users with existing ~/.codex/config.toml (global config) will see "Settings not found" after this change. **Simply click AutoConfigure in LLM Tool Settings** to regenerate the config at {projectRoot}/.codex/config.toml. The old global config can be safely deleted.

## Related

- Follow-up to #609
- Addresses code review feedback (missing tests, stale comments)

# Align Codex Classification and Add Regression Tests

## Summary

This PR aligns documentation and classification for Codex after migrating Codex configuration from a global location (~/.codex/config.toml) to a project-level location ({projectRoot}/.codex/config.toml). It reclassifies Codex as a CLI / project-level (relative-path) editor, renames an internal parameter to reflect project-relative server paths, and adds a regression test suite to verify path handling and configuration resolution.

### McpServerConfigFactory.cs
- Reclassified Codex from desktop/absolute-path to CLI / project-level/relative-path.
- `GetServerPathForEditor` now treats Codex (alongside Claude Code and Gemini CLI) as using relative paths via `ConvertToRelativePath`.
- Desktop editors (Cursor, VSCode, Windsurf) remain the only types that receive the absolute path directly.

### CodexTomlConfigService.cs
- Updated class documentation to reference {projectRoot}/.codex/config.toml (project-level) instead of ~/.codex/config.toml (global).
- Renamed internal BuildBlock parameter from `serverAbsolutePath` → `serverPath` to reflect use of project-relative paths.
- AutoConfigure now uses `MakeRelativeToConfigurationRoot` for server path when writing the project-level config.

### UnityMcpPathResolver.cs
- Updated documentation to list Codex under "CLI / project-level editors (Claude Code, Gemini CLI, Codex)".
- Added `using System.Linq` to support internal path concatenation logic.

### Assets/Tests/Editor/CodexConfigTests.cs (New)
- Added 12 NUnit regression tests covering:
  - BuildBlock / ReadCurrentValues round-trips for relative paths with forward and backslashes, and port preservation.
  - NormalizeForCompare behavior (backslash → forward slash, null, empty).
  - MakeRelativeToConfigurationRoot behavior for paths under project root, outside root, and null/empty inputs.
  - Codex config path/directory resolution under project root (.codex/config.toml and .codex directory).
  - `McpServerConfigFactory` classification: `GetServerPathForEditor` returns relative path for Codex.
- Tests access internal/private static methods via reflection (BindingFlags.NonPublic | BindingFlags.Static) to validate internal logic without changing public APIs.

## Migration Impact

Existing global ~/.codex/config.toml will no longer be discovered. Users should run AutoConfigure in LLM Tool Settings to generate {projectRoot}/.codex/config.toml for project-scoped configuration. The old global config can be removed.

## Test Results

All 12 regression tests pass, validating:
- Configuration serialization/deserialization round-trips.
- Path normalization and separator handling.
- Correct conversion to project-relative paths for in-root servers and preservation of absolute paths for out-of-root servers.
- Configuration path resolution scoped to project root.
- Correct editor classification and relative-path handling for Codex.

### CODE REVIEW

File: Assets/Tests/Editor/CodexConfigTests.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        [Test]
+        public void GetCodexConfigPath_Should_ResolveUnderProjectRoot()
+        {
+            string configPath = UnityMcpPathResolver.GetCodexConfigPath();
+            string projectRoot = UnityMcpPathResolver.GetProjectRoot();
+            string homeDir = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+
+            Assert.IsTrue(configPath.StartsWith(projectRoot),
+                $"Codex config should be under project root, but was: {configPath}");
+            Assert.IsFalse(configPath.StartsWith(homeDir) && !projectRoot.StartsWith(homeDir.TrimEnd('/','\\') + "/"),
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Assets/Tests/Editor/CodexConfigTests.cs]: Good catch! Simplified the assertion to use `Path.Combine` + `Assert.AreEqual` instead of the fragile string manipulation. Fixed in de91919.
